crates/editor/src/editor_tests.rs

```diff
@@ -15047,3 +15049,103 @@ fn assert_hunk_revert(
     cx.assert_editor_state(expected_reverted_text_with_selections);
     assert_eq!(reverted_hunk_statuses, expected_not_reverted_hunk_statuses);
 }
+
+#[gpui::test]
+async fn test_pulling_diagnostics(cx: &mut TestAppContext) {
+    init_test(cx, |_| {});
+
+    let mut cx = EditorLspTestContext::new_rust(
+        lsp::ServerCapabilities {
+            diagnostic_provider: Some(lsp::DiagnosticServerCapabilities::Options(
+                lsp::DiagnosticOptions {
+                    identifier: Some("rust-analyzer".into()),
+                    inter_file_dependencies: true,
+                    workspace_diagnostics: true,
+                    work_done_progress_options: Default::default(),
+                },
+            )),
+            ..Default::default()
+        },
+        cx,
+    )
+    .await;
+
+    let diagnostic_requests = Arc::new(AtomicUsize::new(0));
+    let counter = diagnostic_requests.clone();
+
+    cx.lsp
+        .handle_request::<lsp::request::DocumentDiagnosticRequest, _, _>(move |params, _| {
+            counter.fetch_add(1, atomic::Ordering::Release);
+            assert_eq!(
+                params.text_document.uri,
+                lsp::Url::from_file_path("/root/dir/file.rs").unwrap()
+            );
+            async move {
+                Ok(lsp::DocumentDiagnosticReportResult::Report(
+                    lsp::DocumentDiagnosticReport::Full(lsp::RelatedFullDocumentDiagnosticReport {
+                        related_documents: None,
+                        full_document_diagnostic_report: lsp::FullDocumentDiagnosticReport {
+                            items: vec![],
+                            result_id: None,
+                        },
+                    }),
+                ))
+            }
+        });
+
+    // Opening file should trigger diagnostics
```

Comment:
I think this happens not because the file just is open, but because Zed initializes the language server and inserts it.
We might want to open another file of the same type later in the test to check this comment's statement.